src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -33,122 +34,126 @@ public override WebSocketState State
         {
             get
             {
-#if FEATURE_WASM_THREADS
-                lock (_thisLock)
+                lock (_lockObject)
                 {
-#endif
-                    if (_innerWebSocket == null || _disposed || (_state != WebSocketState.Connecting && _state != WebSocketState.Open && _state != WebSocketState.CloseSent))
+                    if (_innerWebSocket == null || _disposed || _state == WebSocketState.Aborted || _state == WebSocketState.Closed)
                     {
                         return _state;
                     }
-#if FEATURE_WASM_THREADS
-                } //lock
-#endif
-
-#if FEATURE_WASM_THREADS
-                return FastState = _innerWebSocket!.SynchronizationContext.Send(static (BrowserWebSocket self) =>
-                {
-                    lock (self._thisLock)
+                    var st = GetReadyState(_innerWebSocket!);
+                    if (st == WebSocketState.Closed || st == WebSocketState.CloseSent)
                     {
-                        return GetReadyState(self._innerWebSocket!);
-                    } //lock
-                }, this);
-#else
-                return FastState = GetReadyState(_innerWebSocket!);
-#endif
+                        if (_closeReceived && _closeSent)
+                        {
+                            st = WebSocketState.Closed;
```

Comment:
Why do we have this condition and the last one? The `st` is already set to `Closed` or `CloseSent` when we reach this block

Reply:
The JS side state doesn't have separate `closeSent` and `closeReceived` states. That's why we have these bools to make sense of it.

Reply:
Oh, I see. The state can be `CloseSend` and we override it to `Closed` and similar on line 55, state can be `Closed` and we override it to `CloseSent`. That's how it is supposed to work, right?

Reply:
Oh, I see. The state can be `CloseSend` and we override it to `Closed` and similar on line 55, state can be `Closed` and we override it to `CloseSent`. That's how it is supposed to work, right?

Reply:
right